### PR TITLE
Add authentication servlet

### PR DESCRIPTION
**Pull request by PaulKovalov**
Here I add a servlet for user authentication using Google Users API

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.google.sps.servlets;
```
Comment (NeilLi1992) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:1]: Some file documentation would be great.

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * GET request support only
+ * - checks if the user is logged in
+ * - if yes:
+ * -- returns status (logged in)
+ * -- returns logout url
+ * -- returns user's email
+ * - if no:
+ * -- returns status (logged out)
+ * -- returns the login url
+ */
```
Comment (NeilLi1992) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:25]: Good!

**Comment by NeilLi1992** [comment]
Btw I cannot launch it now as it's complaining `package com.google.appengine.api.users does not exist`.

**Comment by PaulKovalov** [comment]
> Btw I cannot launch it now as it's complaining `package com.google.appengine.api.users does not exist`.

Yes, as this branch is based on master which does not have required dependencies in pom.xml yet.

File: portfolio/src/main/webapp/comments.js [hunk 3]
```diff
@@ -21,6 +25,9 @@ async function fetchComments() {
     addDeleteButton();
     document.getElementById('no-comments').remove();
   }
+  // update page
+  updateReplyFromAccessBasedOnAuthState();
```
Comment (harry-7) [inline portfolio/src/main/webapp/comments.js:29]: Is there a reason, we want to create elements and hide them based on authstate? 
Why do we not wanna do the otherway. I.e create elements only when logged in?

Comment (PaulKovalov) [reply portfolio/src/main/webapp/comments.js:29]: well, this particular form is hardcoded in html. To fix this, I need to dynamically create it. I agree, your approach seems more logical

**Pull request closed without merging**